### DISCUSSION TITLE
plugin-cloud-storage: S3 KMS encryption not supported

### DISCUSSION THREAD

**Discussion by jakub-pietrzak** [payloadcms/payload#5596] [Feature Requests & Ideas]
The plugin does not seem to support uploads into S3 buckets that have SSE KMS encryption enabled.

I had to explicitly override the logic for `getHandleUpload` function.

```
    const kmsKeyId = process.env.S3_KMS_KEY_ID
    const sse = kmsKeyId ? ServerSideEncryption.aws_kms : ServerSideEncryption.AES256

    if (file.buffer.length > 0 && file.buffer.length < multipartThreshold) {
      await getStorageClient().putObject({
        Bucket: bucket,
        Key: fileKey,
        Body: fileBufferOrStream,
        ACL: acl,
        ServerSideEncryption: sse, // new param
        SSEKMSKeyId: kmsKeyId, // new param
        ContentType: file.mimeType
      })

      return data
    }
```

That KMS key is set as a default encryption key for the bucket, I'm not sure if it matters, though.

**Comment by paulpopus** [comment]
This plugin is now being maintained in the [Packages Directory](https://github.com/payloadcms/payload/tree/main/packages/plugin-cloud-storage) of the [Payload Monorepo](https://github.com/payloadcms/payload). This repo will soon be archived and all open issues will be closed. This issue, however, will be transferred over.

**Comment by denolfe** [comment]
Moving this to Discussions as a feature request.

**Comment by coldman333** [comment]
For previous two project i add  params to config for plugin
`   ...
      region: process.env.S3_REGION,
      // ... Other S3 configuration
      // @ts-ignore
      params: { ServerSideEncryption: 'AES256' },
      `
      
      and copi plugin code to folders  plugins , and, add this  params to upload request 
      
      `
    const parallelUploadS3 = new Upload({
      client: getStorageClient(),
      params: {
        ACL: acl,
        Body: fileBufferOrStream,
        Bucket: bucket,
        ContentType: file.mimeType,
        Key: fileKey,
        ...params
      },
      partSize: multipartThreshold,
      queueSize: 4,
    })`
    
    today in new  project  i  found some TYPEs conflict  on my code,  and I unhappy support my version of plagin
    
    if it possible add support this  "params" or 'customParams'  or 'actionOptions' to make possible sent it with upload object ?
    
   in Strapi CMS  they use  actionOptions directly to actions
   
   ` actionOptions: {
          upload: {
            ServerSideEncryption: 'AES256',  //fix upload to S3
          },
          uploadStream: {
            ServerSideEncryption: 'AES256', //fix upload to S3
          },
          delete: {},
        },`
